Small fixes & changes

slprime opened the pull request.

slprime commented:
issue: https://github.com/GTNewHorizons/NotEnoughItems/issues/524

OneEyeMaker commented on src/main/java/codechicken/nei/ClientHandler.java:
```diff
@@ -235,7 +235,6 @@ public static void load() {
 
     public static void postInit() {
         loadHandlerOrdering();
-        loadPluginsList();
```
Please explain why you moved this method to a later loading stage.

slprime replied:
this is necessary to fix this problem: https://github.com/GTNewHorizons/NotEnoughItems/issues/524
Immersive Engineering breaks if you load plugins in postInit

OneEyeMaker commented on src/main/java/codechicken/nei/recipe/GuiOverlayButton.java:
```diff
@@ -151,20 +183,16 @@ public void mouseReleased(int mouseX, int mouseY) {
     public void drawItemPresenceOverlay() {
         final int presenceOverlay = NEIClientConfig.itemPresenceOverlay();
         final boolean highlightPresentItem = NEIClientConfig.isSlotHighlightPresent();
-        final List<PositionedStack> ingredients = this.handler.getIngredientStacks(recipeIndex);
-        final List<Boolean> itemPresenceCache = ingredientsOverlay();
+        final ItemOverlayFormat format = presenceOverlay == 2 ? ItemOverlayFormat.BACKGROUND : ItemOverlayFormat.ICON;
```
Instead of hardcoding conversion logic somewhere in implementation, it's better to create static method in `ItemOverlayFormat` enum and hide implementation details:
```java
public static ItemOverlayFormat from(int overlay) {
	return presenceOverlay == 2 ? ItemOverlayFormat.BACKGROUND : ItemOverlayFormat.ICON;
}
```

And then use it here:
```java
final ItemOverlayFormat format = ItemOverlayFormat.from(presenceOverlay);
```

slprime replied:
done

OneEyeMaker reviewed (changes requested):
Nice cleanup overall.
Please, list and explain all your changes for clarification (movement of some method calls is unclear).

slprime commented:
> Nice cleanup overall. Please, list and explain all your changes for clarification (movement of some method calls is unclear).

1. Fixed display of crosses/check marks in bookmarks
![image](https://github.com/user-attachments/assets/495026be-ea91-45d6-817a-289d1939dc28)
2. Hide hotkeys if it not set
3. Search items by original name and renamed name
![image](https://github.com/user-attachments/assets/4c8575d0-393b-4567-81d2-a270f45a2891)
4. Fix shift + click in subsets
![image](https://github.com/user-attachments/assets/1cc48f56-1353-4bf4-93e8-7255ed0b0598)
5. Add ability extend recipe overlay states. (ex. show if item can crafting)
![image](https://github.com/user-attachments/assets/2fc99138-9071-4853-9888-1a85d820b6ac)

P.S.
I rewrote getKeyName to simplify the way it derives neiKeyBindings. And correctly it should be in utilities, not in the config

OneEyeMaker reviewed (approved):
Looks good to me.
